crates/mpz-core/src/block.rs

```diff
@@ -142,13 +142,15 @@ impl Block {
         (self.0[0] & 1) == 1
     }
 
-    /// Let `x0` and `x1` be the lower and higher halves of `x`, respectively.
     /// This function compute ``sigma( x = x0 || x1 ) = x1 || (x0 xor x1)``.
     #[inline(always)]
     pub fn sigma(a: Self) -> Self {
-        let mut x: [u64; 2] = bytemuck::cast(a);
-        x[0] ^= x[1];
-        bytemuck::cast([x[1], x[0]])
+        let val = u128::from_le_bytes(a.0);
+        let low = (val & u64::MAX as u128) as u64;
+        let high = (val >> 64) as u64;
+        let new_low = low ^ high;
+        let result_val = ((new_low as u128) << 64) | (high as u128);
+        Block(result_val.to_le_bytes())
```

Comment:
Is there an advantage in doing the shifting manually instead of using `zerocopy` like `bytemuck` has been used before here?